pkg/dceerr/error.go

```diff
@@ -0,0 +1,208 @@
+package dceerr
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"io"
+	"net/http"
+)
+
+const (
+	clientError        = "ClientError"
+	serverError        = "ServerError"
+	validationError    = "RequestValidationError"
+	alreadyExistsError = "AlreadyExistsError"
+	notFoundError      = "NotFoundError"
+	conflictError      = "ConflictError"
+)
+
+type detailError struct {
+	Message string `json:"message"`
+	Code    string `json:"code"`
+}
+
+// StatusError is the custom error type we are using.
+// Should satisfy errors interface
+type StatusError struct {
```

Comment:
Just a naming thought -- `Status` error is actually a model of an JSON API response. 
But the type doesn't actually include information about the HTTP _status_ code.
Maybe `APIError` would be more suitable?

Reply:
> But the type doesn't actually include information about the HTTP status code.

Oh, I'm wrong. The `apiCode` is the http status code.
I just got confused because our response object also has a `Code` property.

What do you think of something like:
```
type APIError struct {
	httpCode int
	cause   error
	Details detailError `json:"error"`
	stack   *stack
}

Reply:
I don't mind that change.